### ISSUE TITLE
Allowing multiple Azure OpenAI clients 

### ISSUE THREAD

**Issue by adanbrownpaca** [dotnet/aspire#5332]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

Because certain models are only available in certain regions in Azure, I have a need to have multiple Azure AI clients on my project. Dall-e models are only available in EAST US, while GTP40 in EASTUS2, etc. At the moment, only one client with one connection string can be created for dependency injection. 

### Describe the solution you'd like

Basically similar solution to registering HttpClients on the project. You define your class with the right client. Any other solution is fine by me.

**Comment by davidfowl** [comment]
You can call AddKeyedAzureOpenAIClient and pass a name, then used [KeyedServices](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-8.0#keyed-services) to resolve the client
